addons/knowledge/models/knowledge_article.py

```diff
@@ -766,12 +768,14 @@ def invite_members(self, partner_ids, permission, send_mail=True):
         if not self.user_can_write:
             raise AccessError(_("You cannot give access to this article as you are not editor."))
         share_partner_ids = partner_ids.filtered(lambda partner: partner.partner_share)
-        article = self.sudo()
+        article = self
         if permission != 'none':
             article._invite_members(share_partner_ids, 'read', send_mail=send_mail)
             article._invite_members(partner_ids - share_partner_ids, permission, send_mail=send_mail)
         else:
             article._invite_members(partner_ids, permission, send_mail=send_mail)
+
+        article.sudo()._get_descendants().filtered(lambda c: not c.user_can_write)._invite_members(partner_ids, 'none')
```

Comment:
add comment to that line: something like
"Prevent the invited user to get access to the children articles the current user has no access to."